### DISCUSSION TITLE
Receive $statusCode pass $code

### DISCUSSION THREAD

**Discussion by spohess** [symfony/symfony#51602] [Q&A] [answered]
Hi guys, i tried to find answers for this and don't find

Why [HttpException](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/HttpKernel/Exception/HttpException.php) receive $statusCode and pass to parent $code?

When a I use this I need to pass the $statusCode and $code?

Thanks for your attention!

**Comment by stof** [comment]
Can you explain a bit more what you mean ?

**Reply by spohess** [reply]
Hi @stof,

For example, I use Laravel and there has an helper [abort()](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Foundation/Application.php#L1222)

the function use HttpException: `throw new HttpException($code, $message, null, $headers);`, but the HttpException receive the $code as $statusCode and receive $code with default 0 and pass to RuntimeException the $code not $statusCode.

When I run a test I always get 0, if I do this `throw new HttpException($code, $message, null, $headers, $code);` works, but it doesn't seem right to me and I wanted to understand why it doesn't pass the $statusCode.

**Reply by stof** [reply, marked as answer]
Well, the status code in the HttpException constructor is meant to be the HTTP status code that should be used when creating a response, which is accessible through `HttpExceptionInterface::getStatusCode`.
This is totally different from the concept of exception code coming from PHP (which is mostly unused in Symfony, using a code `0` for most the exceptions we trigger, but we still allow other code to use it if they want).

The fact that **Laravel** exposes a function with an argument named `$code` but which actually corresponds to the status code of the HttpException is something that should be reported to Laravel as being confusing, not to Symfony.

**Reply by spohess** [reply]
Thanks for the answer, initially I was going to ask this question in the Laravel discussion, but to better understand the origin I asked it here first.

Thank you very much!
